module-2/backend/week_three.md

```diff
@@ -1,25 +1,64 @@
 
 Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
 Note: When you're done, submit a PR with a reflection in the comments about how this exercise went for you.
 
 ### Questions
 
-1. What are the 3 main features in an ERD?
-2. Create an ERD for the following database: Restaurants, Customers, Items, Ingredients, Beverages, Orders, Vendors.
-3. What is the entry at the command line to create a new rails app?
-4. What do Models generally inherit from in rails?
-5. What do Controllers generally inherit from in a rails project?
-6. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
-7. What rake task is useful when looking at routes, and what information does it give you?
-8. What is an example of a route helper? When would you use them?
-9. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
-10. What are strong params and why are the necessary?
-11. What role does `form_for` play in helping us create our forms?
-12. How does `form_for` know where to submit the user's input?
-13. Create a form using a `form_for` helper to create a new `Horse`. 
-14. Why do we want to validate our models?
+1. What is the entry at the command line to create a new rails app?
+
+    A: rails new app_name
+    
+2. What do Models generally inherit from in rails?
+  
+    A: ActiveRecord::Base
+  
+3. What do Controllers generally inherit from in a rails project?
+
+    A: ApplicationController
+    
+4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+
+    A: resources :horses, only: [:show]
+    
+5. What rake task is useful when looking at routes, and what information does it give you?
+
+    A: rake routes, it shows you a list of the routes that are available to you along with their paths.
+    
+6. What is an example of a route helper? When would you use them?
+
+    A: For the route in question 4, you would use horse_path(horse) as a route helper. They are helpful when you want to  
+      redirect or link within your app and are using RESTful conventions.
+      
+7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+
+    A: `_url` returns the full URL, `_path` only returns the relative path.
+    
+8. What are strong params and why are the necessary?
+
+    A: Strong params allow you to choose params that can be used in controller methods.
+    
+9. What role does `form_for` play in helping us create our forms?
+    
+    A: It makes it easier to make forms in general, and it also allows us to use the same partial for "new" and "edit".
+    
+10. How does `form_for` know where to submit the user's input?
```

Comment:
`form_for` uses the argument that we passed to it (often an instance variable holding either a new or existing instance of an object) to determine the route and verb that will be used in the HTML of the form it generates. I think sometimes this feels a little bit like magic, so next time you get a chance inspect the raw HTML of a form in one of your rails apps. 👍